File: lab spring25/assignment6/taskB.cpp
#include <iostream>
#include <vector>
#include <queue>
using namespace std;

#define rept(i,a,b) for(int i=a;i<b;i++)
#define vi vector<int>
#define vb vector<bool>
#define br cout << '\n'

void makeGraph(vi *graph, int m){
    int u,v;
    while(m--){
        cin >> u >> v;
        graph[u].push_back(v);
        graph[v].push_back(u);
    }
}

int bfs(vi *graph, vi &colr, int s){
    colr[s] = 1;
    int hum=1,rob=0;
    queue<int> Q; Q.push(s);
    int clr = 1;
    while(!Q.empty()){
        int at = Q.front(); Q.pop();
        for(int u:graph[at]) if(colr[u]==-1){
            colr[u] = 1-colr[at];
            colr[u]==1?hum++:rob++;
            Q.push(u);
        }
    }
    return max(hum,rob);
}

void solve(){
    int n,m;
    cin >> n >> m;
    vi graph[n+1];
    makeGraph(graph,m);
    vi colr(n+1,-1);
    int total = 0;
    rept(i,1,n+1) if(colr[i]==-1){
        total+=bfs(graph,colr,i);
    }
    cout << total;br;
}

int main(){
    ios::sync_with_stdio(false);cin.tie(nullptr);cout.tie(nullptr);
    freopen("input.txt","r",stdin);
    int t{1};
    cin >> t;
    while (t--) solve();
}

File: lab summer25/assignment4/taskB.py
import sys
sys.stdin = open("input.txt","r")

def solve():
    n,m = list(map(int,input().split()))
    u = list(map(int,input().split()))
    v = list(map(int,input().split()))
    w = list(map(int,input().split()))
    mat = [[] for _ in range(n+1)]
    for i in range(m): mat[u[i]].append((v[i],w[i]))
    for i in range(1,n+1):
        print(f"{i}: ",end="")
        for e in mat[i]: print(e,end=' ')
        print()

t = 1
t = int(input())
while t>0:
    t-=1
    solve()
    print()